libpermian/plugins/anaconda_webui/__init__.py

```diff
@@ -401,7 +402,8 @@ def _start_vm(self):
         self.virt_install_log = self.crc.openLogfile('virt-install', 'wb', True)
 
         LOGGER.info('Running: ' + repr(cmd))
-        self.proc_virtinstall = subprocess.Popen(cmd, stdout=self.virt_install_log, stderr=subprocess.STDOUT)
+        _, pty_slave = pty.openpty()
```

Comment:
I have some concerns here but it seems like only one of them will be addressable.

1. This means that there will be 2 more file descriptors opened for the whole execution of the workflow which can pose a risk of running out of file descriptors (there is a limit of 1024 file descriptors per process). The way how this could be resolved is to have the pty.openpty somehow called within the preexec_fn which is called in the new process and not in the main permian process. (I don't know now how to achieve this).
2. The file descriptors are by default inherited by the new processes unless close_fds is true and it's up to the child processes how they will handle those inherited descriptors. (this is more of a generic issue rather then related to this specific commit). https://docs.python.org/3/library/os.html#fd-inheritance
3. The file descriptors should be closed when they are not needed (after virt-install is not used).

I know I've suggested this way to resolve this issue, but at that moment, I didn't realize those issue. If it's possible to move the stdin pty handling to preexec_fn, I believe the most problematic issues will be resolved.

Reply:
So far I didn't find a way to utilize preexec_fn to fix this. The pty is now closed in teardown. I don't think we have to worry about running out of fds, the whole pipeline with one workflow running uses 8 and we can always raise the limit in system.